internal/http/parser/errors.go

```diff
@@ -0,0 +1,72 @@
+package http_parser
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/codecrafters-io/http-server-tester/internal/http/inspectable_byte_string"
+)
+
+type IncompleteInputError struct {
+	Reader  *bytes.Reader
+	Message string
+}
+
+type BadProtocolError struct {
```

Comment:
Weird name - can this just be `InvalidHTTPResponseError` and `IncompleteHTTPResponseError`? 

Unlike in Redis we don't have `Decode` and `DecodeRDBFile`, so we don't need the ambiguity here.